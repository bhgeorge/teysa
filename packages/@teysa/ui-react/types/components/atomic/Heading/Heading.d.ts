import { WithHeadingLevel } from './HeadingTag';
declare type HeadingSize = 'xs' | 'sm' | 'md' | 'lg' | 'xl';
interface HeadingProps extends WithHeadingLevel {
    size?: HeadingSize;
}
declare const Heading: ({ headingLevel, headingOffset, children, size }: HeadingProps) => JSX.Element;
export default Heading;
