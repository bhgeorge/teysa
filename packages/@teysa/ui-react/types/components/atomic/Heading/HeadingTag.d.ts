import { ReactNode } from 'react';
declare type ValidHeadingLevel = 1 | 2 | 3 | 4 | 5 | 6;
export interface WithHeadingLevel {
    children: ReactNode | ReactNode[];
    className?: string;
    headingLevel?: ValidHeadingLevel;
    headingOffset?: 0 | ValidHeadingLevel;
}
declare const HeadingTag: ({ children, className, headingLevel, headingOffset, }: WithHeadingLevel) => JSX.Element;
export default HeadingTag;
