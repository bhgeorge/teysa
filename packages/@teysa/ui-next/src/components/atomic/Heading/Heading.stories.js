// Local
import { Heading } from './Heading';

export default {
  title: 'Atomic/Heading',
  component: Heading,
};

const Template = props => <Heading {...props}>Teysa Karlov</Heading>;

export const Default = Template.bind({});

const TemplateWithContext = props => (
  <div className="max-w-prose">
    <Heading {...props}>Teysa Karlov</Heading>
    <p className="mb-4">
      If a creature dying causes a triggered ability of a permanent you control
      to trigger, that ability triggers an additional time.
    </p>
    <p className="mb-4">
      Creature tokens you control have vigilance and lifelink.
    </p>
    <p>
      <i>
        While Kaya is guildmaster, Teysa waits patiently at the center of the
        Orzhov web of power.
      </i>
    </p>
  </div>
);

export const WithContext = TemplateWithContext.bind({});
