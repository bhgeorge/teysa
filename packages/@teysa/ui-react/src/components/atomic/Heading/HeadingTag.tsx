// Global
import React, { ReactNode } from 'react';

type ValidHeadingLevel = 1 | 2 | 3 | 4 | 5 | 6;

export interface WithHeadingLevel {
  children: ReactNode | ReactNode[];
  className?: string;
  headingLevel?: ValidHeadingLevel;
  headingOffset?: 0 | ValidHeadingLevel;
}

const TAGS = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6'];

const HeadingTag = ({
  children,
  className,
  headingLevel = 2,
  headingOffset = 0,
}: WithHeadingLevel) => {
  const i = headingLevel + headingOffset - 1;
  const tag = i >= TAGS.length ? 'p' : TAGS[i];
  const HTag = tag as keyof JSX.IntrinsicElements;

  return <HTag className={className}>{children}</HTag>;
};

export default HeadingTag;
