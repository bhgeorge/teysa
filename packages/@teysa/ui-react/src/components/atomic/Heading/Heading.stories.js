// Local
import Heading from './Heading';

export default {
  title: 'Atomic/Heading',
  component: Heading,
};

const Template = (props) => <Heading {...props}>Teysa Karlov</Heading>;

export const Default = Template.bind({});
