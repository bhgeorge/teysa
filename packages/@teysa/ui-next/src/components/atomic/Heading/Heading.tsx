// Global
import React from 'react';
import classNames from 'classnames';
// Local
import { HeadingTag, WithHeadingLevel } from './HeadingTag';

type HeadingSize = 'xs' | 'sm' | 'md' | 'lg' | 'xl';

interface HeadingProps extends WithHeadingLevel {
  size?: HeadingSize;
}

const sizeClasses: Record<HeadingSize, string> = {
  xs: 'text-sm font-bold mb-2',
  sm: 'text-xl md:text-2xl font-bold mb-4',
  md: 'text-2xl md:text-4xl font-bold mb-4',
  lg: 'text-4xl md:text-5xl font-bold leading-snug md:leading-snug mb-8',
  xl: 'text-6xl md:text-7xl font-bold leading-snug md:leading-snug mb-16',
};

export function Heading({
  headingLevel,
  headingOffset,
  children,
  size = 'md',
}: HeadingProps) {
  return (
    <HeadingTag
      headingLevel={headingLevel}
      headingOffset={headingOffset}
      className={classNames(sizeClasses[size])}
    >
      {children}
    </HeadingTag>
  );
}
