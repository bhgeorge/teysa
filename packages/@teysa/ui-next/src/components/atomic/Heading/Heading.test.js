// Global
import React from 'react';
import { render } from '@testing-library/react';
// Local
import { Heading } from './Heading';

const snapshot = size => {
  const component = render(<Heading size={size}>Teysa</Heading>);
  expect(component.container.firstChild).toMatchSnapshot();
};

it('renders at each size', () => {
  ['xs', 'sm', 'md', 'lg', 'xl'].forEach(size => {
    snapshot(size);
  });
});
