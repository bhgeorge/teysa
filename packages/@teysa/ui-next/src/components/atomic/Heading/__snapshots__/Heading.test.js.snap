// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders at each size 1`] = `
<h2
  class="text-sm font-bold mb-2"
>
  Teysa
</h2>
`;

exports[`renders at each size 2`] = `
<h2
  class="text-xl md:text-2xl font-bold mb-4"
>
  Teysa
</h2>
`;

exports[`renders at each size 3`] = `
<h2
  class="text-2xl md:text-4xl font-bold mb-4"
>
  Teysa
</h2>
`;

exports[`renders at each size 4`] = `
<h2
  class="text-4xl md:text-5xl font-bold leading-snug md:leading-snug mb-8"
>
  Teysa
</h2>
`;

exports[`renders at each size 5`] = `
<h2
  class="text-6xl md:text-7xl font-bold leading-snug md:leading-snug mb-16"
>
  Teysa
</h2>
`;
